src/test/java/com/example/onlinebookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,282 @@
+        CategoryDto actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                CategoryDto.class);
+        boolean isEquals = EqualsBuilder.reflectionEquals(expectedCategoryDto, actual, "id");
+        assertTrue(isEquals);
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Sql(scripts = {
+            "classpath:db/book-category-insert.sql"
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:db/book-category-delete.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Test update with valid request")
+    @Test
+    void updateCategory_validCategory_returnsUpdatedCategory() {
+        CreateCategoryRequestDto updatedRequestDto = requestDto;
+        updatedRequestDto.setName("updated Name");
+        String jsonRequest = objectMapper.writeValueAsString(updatedRequestDto);
+        MvcResult result = mockMvc.perform(put("/api/categories/1")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(updatedRequestDto)))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn();
+        CategoryDto actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                CategoryDto.class);
+        boolean isEquals = EqualsBuilder.reflectionEquals(
+                expectedCategoryResponseDto, actual, "id");
+        assertTrue(isEquals);
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Sql(scripts = {
+            "classpath:db/book-category-insert.sql"
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:db/book-category-delete.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Test delete with valid request")
+    @Test
+    void deleteCategoryById_validId_deleteCategory() {
+        mockMvc.perform(delete("/api/categories/2")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNoContent())
+                .andReturn();
+        assertThrows(EntityNotFoundException.class,
+                () -> categoryRepository.findById(2L).orElseThrow(
+                        () -> new EntityNotFoundException("Can't get category by id: 2")));
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Sql(scripts = {
+            "classpath:db/book-category-insert.sql"
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:db/book-category-delete.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Test deleteById with non valid request")
+    @Test
+    void deleteById_NonValidId_ThrowsException() {
```

Comment:
```suggestion
    void deleteById_nonValidId_throwsException() {
```
make sure you are fixed it in all methods